src/Microsoft.ML.Data/TrainCatalog.cs

```diff
@@ -20,7 +20,7 @@ namespace Microsoft.ML
     /// </summary>
     public abstract class TrainCatalogBase
     {
-        protected internal readonly IHost Host;
+        protected internal IHostEnvironment Host { get; set; }
 
         [BestFriend]
         internal IHostEnvironment Environment => Host;
```

Comment:
I don't see the point of having both. Can we just this, as an auto-implemented property?

Reply:
did u mean  ".. just *use* this" ?  

Removed the `Host` property, and using `Environment` instead. 

---
In reply to: [256217044](https://github.com/dotnet/machinelearning/pull/2524#discussion_r256217044) [](ancestors = 256217044)